app/src/main/java/ru/suslovalex/androidacademyapp/viewmodel/MoviesListViewModel.kt

```diff
@@ -1,37 +1,49 @@
 package ru.suslovalex.androidacademyapp.viewmodel
 
-import android.content.Context
+import android.util.Log
 import androidx.lifecycle.*
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
-import ru.suslovalex.androidacademyapp.data.Movie
-import ru.suslovalex.androidacademyapp.data.loadMovies
+import ru.suslovalex.androidacademyapp.repository.Repository
+import ru.suslovalex.androidacademyapp.data.GenreResponse
+import ru.suslovalex.androidacademyapp.data.MoviesResponse
 import ru.suslovalex.androidacademyapp.domain.MovieChecker
 import ru.suslovalex.androidacademyapp.domain.MovieResponseResult
+import ru.suslovalex.androidacademyapp.retrofit.RetrofitInstance
 
-class MoviesListViewModel(private val checker: MovieChecker) : ViewModel() {
+class MoviesListViewModel(private val checker: MovieChecker, private val repository: Repository) : ViewModel() {
 
     private val _state = MutableLiveData<MoviesListState>()
     val moviesListState: LiveData<MoviesListState>
         get() = _state
 
+    private lateinit var genreResponse: GenreResponse
 
-    fun getMovies(context: Context) = viewModelScope.launch(Dispatchers.IO) {
+    init {
+        getGenres()
```

Comment:
@AleksandrAlex Also not a very good solution. You need to map all the data inside repository and only call repository method in ViewModel